# Challenge 3

In this challenge we will work on the `Orders` data set. In your work you will apply the thinking process and workflow we showed you in Challenge 2.

You are serving as a Business Intelligence Analyst at the headquarter of an international fashion goods chain store. Your boss today asked you to do two things for her:

**First, identify two groups of customers from the data set.** The first group is **VIP Customers** whose **aggregated expenses** at your global chain stores are **above the 95th percentile** (aka. 0.95 quantile). The second group is **Preferred Customers** whose **aggregated expenses** are **between the 75th and 95th percentile**.

**Second, identify which country has the most of your VIP customers, and which country has the most of your VIP+Preferred Customers combined.**

## Q1: How to identify VIP & Preferred Customers?

We start by importing all the required libraries:

In [1]:
# import required libraries
import numpy as np
import pandas as pd

Next, extract and import `Orders` dataset into a dataframe variable called `orders`. Print the head of `orders` to overview the data:

In [7]:
# your code here
orders = pd.read_csv('Orders.csv')
orders.head(10)

Unnamed: 0  InvoiceNo StockCode  year  month  day  hour  \
0           0     536365    85123A  2010     12    3     8   
1           1     536365     71053  2010     12    3     8   
2           2     536365    84406B  2010     12    3     8   
3           3     536365    84029G  2010     12    3     8   
4           4     536365    84029E  2010     12    3     8   
5           5     536365     22752  2010     12    3     8   
6           6     536365     21730  2010     12    3     8   
7           7     536366     22633  2010     12    3     8   
8           8     536366     22632  2010     12    3     8   
9           9     536367     84879  2010     12    3     8   

                           Description  Quantity          InvoiceDate  \
0   white hanging heart t-light holder         6  2010-12-01 08:26:00   
1                  white metal lantern         6  2010-12-01 08:26:00   
2       cream cupid hearts coat hanger         8  2010-12-01 08:26:00   
3  knitted union flag hot water bottle         6  2010-12-01 08:26:00   
4       red woolly hottie white heart.         6  2010-12-01 08:26:00   
5         set 7 babushka nesting boxes         2  2010-12-01 08:26:00   
6    glass star frosted t-light holder         6  2010-12-01 08:26:00   
7               hand warmer union jack         6  2010-12-01 08:28:00   
8            hand warmer red polka dot         6  2010-12-01 08:28:00   
9        assorted colour bird ornament        32  2010-12-01 08:34:00   

   UnitPrice  CustomerID         Country  amount_spent  
0       2.55       17850  United Kingdom         15.30  
1       3.39       17850  United Kingdom         20.34  
2       2.75       17850  United Kingdom         22.00  
3       3.39       17850  United Kingdom         20.34  
4       3.39       17850  United Kingdom         20.34  
5       7.65       17850  United Kingdom         15.30  
6       4.25       17850  United Kingdom         25.50  
7       1.85       17850  United Kingdom         11.10  
8       1.85       17850  United Kingdom         11.10  
9       1.69       13047  United Kingdom         54.08

---

"Identify VIP and Preferred Customers" is the non-technical goal of your boss. You need to translate that goal into technical languages that data analysts use:

## How to label customers whose aggregated `amount_spent` is in a given quantile range?


We break down the main problem into several sub problems:

#### Sub Problem 1: How to aggregate the  `amount_spent` for unique customers?

#### Sub Problem 2: How to select customers whose aggregated `amount_spent` is in a given quantile range?

#### Sub Problem 3: How to label selected customers as "VIP" or "Preferred"?

*Note: If you want to break down the main problem in a different way, please feel free to revise the sub problems above.*

Now in the workspace below, tackle each of the sub problems using the iterative problem solving workflow. Insert cells as necessary to write your codes and explain your steps.

In [12]:
orders.shape

(397924, 14)

In [10]:
# your code here
orders['CustomerID'].unique()

array([17850, 13047, 12583, ..., 13298, 14569, 12713], dtype=int64)

In [11]:
len(orders['CustomerID'].unique())

4339

We have 4339 customers in total while we have 397924 orders

In [32]:
orders_by_customer = orders.groupby('CustomerID')['amount_spent'].sum()
display(orders_by_customer)

CustomerID
12346    77183.60
12347     4310.00
12348     1797.24
12349     1757.55
12350      334.40
           ...   
18280      180.60
18281       80.82
18282      178.05
18283     2094.88
18287     1837.28
Name: amount_spent, Length: 4339, dtype: float64

We now have the spent by customer. We can see that the lenght is 4339 has we talked earlier

In [37]:
condition_vip = orders_by_customer > np.quantile(orders_by_customer,.95)
vip_customers = orders_by_customer[condition_vip]
vip_customers

CustomerID
12346     77183.60
12357      6207.67
12359      6372.58
12409     11072.67
12415    124914.53
           ...    
18109      8052.97
18139      8438.34
18172      7561.68
18223      6484.54
18229      7276.90
Name: amount_spent, Length: 217, dtype: float64

Now we know the VIP Customers, 217 customers in total

In [38]:
condition_prefered_1 = orders_by_customer > np.quantile(orders_by_customer,.75)
condition_prefered_2 = orders_by_customer < np.quantile(orders_by_customer,.95)
prefered_customers = orders_by_customer[condition_prefered_1 & condition_prefered_2]
prefered_customers

CustomerID
12347    4310.00
12348    1797.24
12349    1757.55
12352    2506.04
12356    2811.43
          ...   
18259    2338.60
18260    2643.20
18272    3078.58
18283    2094.88
18287    1837.28
Name: amount_spent, Length: 868, dtype: float64

And now we know the prefered Customers, 868 customers in total

In order to laber the customer, we will create a new column:
Depending on the spend of the customer, he will belong to VIP, prefered, or null

In [ ]:
orders['group'] =

Now we'll leave it to you to solve Q2 & Q3, which you can leverage from your solution for Q1:

## Q2: How to identify which country has the most VIP Customers?

De la question au dessus, je groupby les VIP et je value count les pays

In [ ]:
# your code here

## Q3: How to identify which country has the most VIP+Preferred Customers combined?

In [ ]:

De la question au dessus, je groupby les VIP et les prefferd et je value count les pays# your code here